drivers/serial/uart_rpi_pico.c

```diff
@@ -302,6 +309,7 @@ static const struct uart_driver_api uart_rpi_driver_api = {
 		.uart_dev = uart##idx,						\
 		.uart_regs = (uart_hw_t *)uart##idx,				\
 		.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(idx),			\
+		.reset = RESET_DT_INST_DEV_CONFIG_INIT(idx, id),		\
```

Comment:
this driver has a dependency on `CONFIG_RESET=y` now, maybe we should specify this in Kconfig. Same can be said on pinctrl and I think most drivers don't do that, though.